lib/connectors/memory.js

```diff
@@ -207,18 +207,23 @@ Memory.prototype.create = function create(model, data, options, callback) {
   var idName = this.idName(model);
   id = (props[idName] && props[idName].type && props[idName].type(id)) || id;
   this.setIdValue(model, data, id);
-  if(!this.collection(model)) {
+  if (!this.collection(model)) {
     this.collection(model, {});
   }
 
-  if (this.collection(model)[id]) {
-    return process.nextTick(function() {
-      callback(new Error('Duplicate entry for ' + model + '.' + idName));
-    });
-  }
+  if (this.collection(model)[id])
+    return callback(new Error('Duplicate entry for ' + model + '.' + idName));
 
   this.collection(model)[id] = serialize(data);
-  this.saveToFile(id, callback);
+  callback(null, id);
+}
+
+Memory.prototype.create = function create(model, data, options, callback) {
+  var self = this;
+  this._create(model, data, function(err, id) {
+    if (err) return callback(err);
```

Comment:
I think you need to call `process.nextTick` here, because `_create` calls the callback immediately (sync).